sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GoneAndRetryWithRetryPolicy.java

```diff
@@ -83,11 +84,10 @@ public GoneAndRetryWithRetryPolicy(RxDocumentServiceRequest request, Integer wai
                                 exception, this.lastRetryWithException);
                         exceptionToThrow = this.lastRetryWithException;
                     } else {
-                        logger.warn("Received gone exception after backoff/retry. Will fail the request. {}",
-                                exception.toString());
+                        logger.warn("Received gone exception after backoff/retry. Will fail the request. {}, short stackTrace = [{}]",
```

Comment:
Why is this logger.warn() and not logger.info(), is this intentional?

Reply:
after all the retries are exhausted we want to print a warn message indicating failure. however when there are more retry attempt are left we want to print a info message.